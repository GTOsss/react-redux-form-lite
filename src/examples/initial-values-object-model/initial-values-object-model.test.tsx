import * as React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import {reducer} from '../../index';
import InitialValues from './initial-values-object-model';

export interface IValues {
  profile?: {
    firstName?: string;
    lastName?: string;
    sex?: string;
  };
  contacts?: {
    email?: string;
  };
  additionalInfo?: {
    favoriteColor?: string;
    employed?: boolean;
    more?: {
      notes?: string;
    }
  };
}

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}>
      <InitialValues onSubmit={onSubmit} />
    </Provider>,
  );

  return {
    reduxThunkTester,
    getFormState: () => store.getState().reduxForm.simple as IReduxFormState<IValues>,
    component,
  };
};

describe('<InitialValuesObjectModel />', () => {
  test('Render inputs', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render inputs (store)', () => {
    const {getFormState} = renderComponent();
    expect(getFormState()).toMatchSnapshot();
  });

  test('Render inputs after change', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'new value 0'}});
    component.find('input').at(1).simulate('change', {target: {value: 'new value 1'}});
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Reset form (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'some name'}});
    component.find('input').at(1).simulate('change', {target: {value: 'some second name'}});
    component.find('input').at(2).simulate('change', {target: {value: 'some email'}});
    component.find('#resetForm').simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });
});
