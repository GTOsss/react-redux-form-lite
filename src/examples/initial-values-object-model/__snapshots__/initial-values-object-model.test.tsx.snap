// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitialValuesObjectModel /> Render inputs (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "initialValues": Object {
    "additionalInfo": Object {
      "employed": true,
      "favoriteColor": "ff0000",
      "more": Object {
        "notes": "text for notes",
      },
    },
    "contacts": Object {
      "email": "test@mail.ru",
    },
    "profile": Object {
      "firstName": "Timofey",
      "lastName": "Goncharov",
    },
  },
  "meta": Object {
    "additionalInfo": Object {
      "favoriteColor": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "more": Object {
        "notes": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
    },
    "contacts": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "employed": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "sex": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "additionalInfo": Object {
      "employed": true,
      "favoriteColor": "ff0000",
      "more": Object {
        "notes": "text for notes",
      },
    },
    "contacts": Object {
      "email": "test@mail.ru",
    },
    "employed": undefined,
    "profile": Object {
      "firstName": "Timofey",
      "lastName": "Goncharov",
      "sex": undefined,
    },
  },
}
`;

exports[`<InitialValuesObjectModel /> Render inputs 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="Timofey"
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="Goncharov"
  />,
  <input
    name="contacts.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="email"
    value="test@mail.ru"
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    id="employed"
    name="employed"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value=""
  />,
]
`;

exports[`<InitialValuesObjectModel /> Render inputs after change 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="new value 0"
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="new value 1"
  />,
  <input
    name="contacts.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="email"
    value="test@mail.ru"
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    id="employed"
    name="employed"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value=""
  />,
]
`;

exports[`<InitialValuesObjectModel /> Reset form (render inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="Timofey"
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="Goncharov"
  />,
  <input
    name="contacts.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="email"
    value="test@mail.ru"
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    name="profile.sex"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value=""
  />,
  <input
    id="employed"
    name="employed"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value=""
  />,
]
`;
